File: car-rental-api/CarRental/src/main/java/CarRental/dto/booking/BookingResponse.java
package CarRental.dto.booking;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

import lombok.Builder;

@Builder
public record BookingResponse(
        Long id,
        Long userId,
        Long carId,
        String renterFirstName,
        String renterLastName,
        LocalDate renterDob,
        String renterEmail,
        String renterPhone,
        Long renterWardId,
        String renterLocation,
        String renterNationalId,
        String renterDriverLicenseImageUrl,
        // driver information
        String driverFirstName,
        String driverLastName,
        LocalDate driverDob,
        String driverEmail,
        String driverPhone,
        Long driverWardId,
        String driverLocation,
        String driverNationalId,
        String driverLicenseImageImageUrl,
        // booking information
        LocalDateTime pickUpDateTime,
        LocalDateTime returnDateTime,
        Integer numberOfDays,
        Long pickUpWardId,
        String pickUpLocation,
        BigDecimal basePrice,
        BigDecimal deposit,
        BigDecimal total,
        String status,
        LocalDateTime createdAt,
        String renterMessage,
        String carOwnerMessage,
        String paymentType) {

}

File: car-rental-portal/Frontend/src/app/models/interfaces/booking.response.ts
import {User} from "../entities/user.model";

export interface BookingResponseBasic {
    id: number;
    userId: number;
    carId: number;
    pickUpDateTime: string;
    returnDateTime: string;
    basePrice: number;
    deposit: number;
    total: number;
    status: string;
    createdAt: string;
    numberOfDays: number;
    renterMessage?: string;
    carOwnerMessage?: string;
    paymentType: string;
    isRating: boolean,
    user?: User,
    carName?: string,
    carImage?: string,
    wardId?: number,
    road?: string,
    actualPickUp?: string;
    actualReturn?: string,
    message?: string
}
